homeassistant/components/zha/device_tracker.py

```diff
@@ -0,0 +1,115 @@
+"""Support for the ZHA platform."""
+import logging
+import time
+from homeassistant.components.device_tracker import (
+    SOURCE_TYPE_ZIGBEE, DOMAIN
+)
+from homeassistant.components.device_tracker.config_entry import (
+    DeviceTrackerEntity
+)
+from homeassistant.const import (
+    STATE_NOT_HOME,
+    STATE_HOME
+)
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from .core.const import (
+    DATA_ZHA, DATA_ZHA_DISPATCHERS, ZHA_DISCOVERY_NEW,
+    POWER_CONFIGURATION_CHANNEL, SIGNAL_STATE_ATTR,
+    SIGNAL_ATTR_UPDATED
+)
+from .entity import ZhaEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Zigbee Home Automation device tracker from config entry."""
+    async def async_discover(discovery_info):
+        await _async_setup_entities(hass, config_entry, async_add_entities,
+                                    [discovery_info])
+
+    unsub = async_dispatcher_connect(
+        hass, ZHA_DISCOVERY_NEW.format(DOMAIN), async_discover)
+    hass.data[DATA_ZHA][DATA_ZHA_DISPATCHERS].append(unsub)
+
+    device_trackers = hass.data.get(DATA_ZHA, {}).get(DOMAIN)
+    if device_trackers is not None:
+        await _async_setup_entities(hass, config_entry, async_add_entities,
+                                    device_trackers.values())
+        del hass.data[DATA_ZHA][DOMAIN]
+
+    return True
+
+
+async def _async_setup_entities(hass, config_entry, async_add_entities,
+                                discovery_infos):
+    """Set up the ZHA device trackers."""
+    entities = []
+    for discovery_info in discovery_infos:
+        entities.append(ZHADeviceTrackerEntity(**discovery_info))
+
+    async_add_entities(entities, update_before_add=True)
+
+
+class ZHADeviceTrackerEntity(DeviceTrackerEntity, ZhaEntity):
+    """Represent a tracked device."""
+
+    def __init__(self, **kwargs):
+        """Initialize the ZHA device tracker."""
+        super().__init__(**kwargs)
+        self._battery_channel = self.cluster_channels.get(
+            POWER_CONFIGURATION_CHANNEL)
+        self._last_seen = None
+        self._seen = False
+        self._keepalive_interval = 60
+        self._should_poll = True
+
+    async def async_added_to_hass(self):
+        """Run when about to be added to hass."""
+        await super().async_added_to_hass()
+        if self._battery_channel:
+            await self.async_accept_signal(
+                self._battery_channel, SIGNAL_STATE_ATTR,
+                self.async_attribute_updated)
+            await self.async_accept_signal(
+                self._battery_channel, SIGNAL_ATTR_UPDATED,
+                self.async_attribute_updated)
+
+    async def async_update(self):
```

Comment:
Instead of relying on polling, use `async_call_later` to automatically update the state to be not home.

Reply:
how would that work? This needs to happen continually. we're checking the last seen on the device in zigpy (local property check... nothing remote) against a threshold. The way the device works is by sending battery reports very frequently once it connects to the network.

Reply:
Something like this:

```python
self.state = STATE_HOME
if self._unsub_state_check:
    self._unsub_state_check()
self._unsub_state_check = async_call_later(self.hass, 10, self._mark_not_home)
```

Reply:
Updated. How about something like this?

Reply:
Polling feels weird if we know the exact time it would be marked as stale.

Reply:
Ok. So is the updated code acceptable?